packages/worker/native/src/lib.rs

```diff
@@ -402,12 +458,12 @@ macro_rules! js_value_getter {
 
 // Below are functions exported to JS
 
-/// Create a new worker asynchronously.
+/// Create a new core instance asynchronously.
 /// Immediately spawns a poller thread that will block on [Request]s
-/// Worker is returned to JS using supplied callback
-fn worker_new(mut cx: FunctionContext) -> JsResult<JsUndefined> {
-    let worker_options = cx.argument::<JsObject>(0)?;
-    let server_options = worker_options
+/// Core is returned to JS using supplied callback
+fn core_new(mut cx: FunctionContext) -> JsResult<JsUndefined> {
```

Comment:
Once you have shared core working properly, I would change this function to use `lazy_static!` or similar so that it is guaranteed there is only ever one core instance -- then this can be called `get_or_instantiate_core` or something.

Reply:
I don't remember what we said about multiple cores in a single process, why can't we have those?